tests/integration/suite/test_catalog.py

```diff
@@ -292,21 +293,25 @@ def test_invalid_catalog_charts(admin_mc, remove_resource):
     name = random_str()
     url = "https://github.com/rancher/integration-test-charts"
 
-    catalog = client.create_catalog(name=name,
-                                    branch="broke-charts",
-                                    url=url,
-                                    )
-    remove_resource(catalog)
-    wait_for_template_to_be_created(client, catalog.id)
-
-    catalog = client.reload(catalog)
+    def get_errored_catalog():
+        catalog = client.create_catalog(name=name,
```

Comment:
This is going to keep trying to create the catalog every time wait_for calls this function, which I don't think is intended. Create the catalog outside this check then do the refresh/check in here